cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginService.java

```diff
@@ -36,35 +37,44 @@
 import com.google.common.cache.RemovalListener;
 import com.google.common.cache.RemovalNotification;
 import com.google.common.io.Closeables;
+import com.google.common.util.concurrent.AbstractService;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.Closeable;
 import java.io.File;
 import java.io.IOException;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
 import javax.ws.rs.Path;
 
 /**
  * To find, instantiate and invoke methods in plugin artifacts
  */
-public class PluginService {
+@Singleton
+public class PluginService extends AbstractService {
```

Comment:
Extend Google's AbstractIdleService.  AbstractService is meant for user programs, and has methods like usePluginClass which make no sense for a system service.